### ISSUE TITLE
Restarting nomad agent forgets about script check

### ISSUE THREAD

**Issue by JanBerktold** [hashicorp/nomad#7857]
### Nomad version
```
root@chi1-lapp1508:~# nomad version
Nomad v0.9.7+ent (cdc3b91b156e1668b733fea28aa27295dc4dbdd6)
```

### Operating system and Environment details
```
root@chi1-lapp1508:~# cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
### Issue

When restarting the Nomad agent on a server that is hosting an allocation with a script check, the restarting agent will pick up the allocation again but "forget" about the script check, this causes the check to expire in Consul and therefore fail the check.

After submitting the job, everything is fine:
![image](https://user-images.githubusercontent.com/3247489/80949721-657e8000-8da9-11ea-90e2-24b63f4beeb0.png)

After restarting the nomad binary, the alloc stays in this state forever:
![image](https://user-images.githubusercontent.com/3247489/80949939-d3c34280-8da9-11ea-9461-92424f2ab26d.png)

### Reproduction steps
1. Submit job spec (attached below)
2. On the server that's running the allocation, restart the agent (`systemctl restart nomad` in my case)
3. Wait a few seconds and check Consul, the script check will have expired (2nd screenshot above)

### Job file (if appropriate)
```
job "script-check-repro" {
  namespace = "caching-platform"
  datacenters = ["chi1-caching"]
  type = "service"

  update {
    max_parallel = 1

    min_healthy_time = "10s"

    healthy_deadline = "3m"

    progress_deadline = "10m"

    auto_revert = false

    canary = 0
  }

  migrate {
    max_parallel = 1

    health_check = "checks"

    min_healthy_time = "10s"

    healthy_deadline = "5m"
  }

  group "cache-repro" {
    count = 1

    restart {
      attempts = 2
      interval = "30m"
      delay = "15s"
      mode = "fail"
    }

    ephemeral_disk {
      size = 300
    }







    task "redis" {
      driver = "docker"

      config {
        image = "redis:3.2"
        port_map {
          db = 6379
        }
      }


      resources {
        network {
          mbits = 10
          port "db" {}
        }
      }

      service {
        name = "redis-cache"
        tags = ["global", "cache"]
        port = "db"
        check {
          name     = "alive"
          type     = "tcp"
          interval = "10s"
          timeout  = "2s"
        }
        check {
          name = "the repro check"
          type = "script"
          command = "/bin/sh"
          args = ["-c", "redis-cli PING"]
          interval = "10s"
          timeout  = "2s"
        }
      }
    }
  }
}
```
### Nomad Client logs (if appropriate)
I could not find any relevant logs when greping for the alloc id, job or task name. Let me know that to look for please.

**Comment by JanBerktold** [comment]
Good news: It looks like `check_restart` is still respected in this state so that's a possible workaround.

**Comment by notnoop** [comment]
Thanks for raising this issue.  This was fixed in 0.10.0 as part of a bigger consul refactor - in https://github.com/hashicorp/nomad/pull/6197 and https://github.com/hashicorp/nomad/pull/6291 .

Can you please retry with a more recent version?

**Comment by robloxrob** [comment]
As of now we are stuck on 0.9, but hopefully we can migrate to 0.10 soon.
